telekom/testerra-hpqc-connector
# Added sync with cucumber tests

# Description

- The annotation `@QCTestname` for Java methods supportes the attribute `testId`. Instead of using the the name you can use the QC test id:
    ````java
    @QCTestname(testId = 2264)
    @Test
    public void testT04a_QcSyncResultPassedSpacesPerId() {
    }
    ````

* Syncing of Cucumber scenarios is supported now: 
  * Use the tag `@QCTestname("QC-Testname")` _or_
  * use the tag `@QCTestId("123")`
  * See an example in the subproject `cucumber-test` or in README.

## Breaking change

The annotation classes `@QCTestset` and `@QCTestname` are now part of QC connector. The classes of Testerra will be ignored! 

For migration please update your imports:

````java
// with old version
import eu.tsystems.mms.tic.testframework.testmanagement.annotation.QCTestname;
import eu.tsystems.mms.tic.testframework.testmanagement.annotation.QCTestset;

// change it to
import eu.tsystems.mms.tic.testframework.qcconnector.annotation.QCTestname;
import eu.tsystems.mms.tic.testframework.qcconnector.annotation.QCTestset;
````

Fixes #5 

## What else?
* Added duration to test runs

## Type of change

Please delete options that are not relevant.


# Checklist:
